src/dmd/traits.d

```diff
@@ -1645,7 +1645,38 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
         if (cmp(s1,s2) || cmp(s2,s1))
             return True();
 
-        return (s1 == s2) ? True() : False();
+        if (s1 == s2)
+            return True();
+
+        // https://issues.dlang.org/show_bug.cgi?id=18771
+        // OverloadSets are equal if they contain the same functions
+        auto overSet1 = s1.isOverloadSet();
+        auto overSet2 = s2.isOverloadSet();
+        if (overSet1 && overSet2)
+        {
+            if (overSet1.a.dim != overSet2.a.dim)
+                return False();
+
+            // OverloadSets contain array of Dsymbols => O(n*n)
+            // to compare for equality as the order of overloads
+            // might not be the same
+            foreach(elem1; overSet1.a)
+            {
+                bool flag;
+                foreach(elem2; overSet2.a)
+                {
+                    if (elem2 == elem1)
+                    {
+                        flag = true;
```

Comment:
Any reason to not just `return False();` here?

Reply:
Even better would be just label the outer loop, then use `continue label;` here. No `flag` is required.